codalab/worker/local_run/local_dependency_manager.py

```diff
@@ -127,6 +127,7 @@ def _load_state(self):
 
     def start(self):
         logger.info('Starting local dependency manager')
+        self._sync_dependency_state()
```

Comment:
A partial version of this logic is implemented on lines 100-120ish of this file. Can you maybe merge that logic into your function and call this function there? (It might be good to log deleted entries etc like we do on lines 100-120 right now)